src/modules/cacheClient/cacheClient.service.ts

```diff
@@ -35,69 +36,80 @@ export class CacheClient implements ICacheClient {
         });
         this.httpClient.interceptors.response.use((response: AxiosResponse) => {
             return response;
-        }, this.handleUnauthorized.bind(this));
+        }, this.handleError.bind(this));
         this.authEnabled = cacheServerSupportsAuth;
         this.isBrowser = executionEnvironment() === ExecutionEnvironment.BROWSER;
+        if (!this.isBrowser) {
+            this.httpClient.defaults.headers.common["Authorization"] = `Bearer ${this.token}`;
+        }
```

Comment:
Why was this portion moved to the `init` method? What if the `token` is refreshed? Doesn't the `Authorization` header need to be refreshed as well?
Is `token` even set at this point 🤔

Reply:
Also, I wonder if we should flag the fact that we are returning the `token` as a possible security vulnerability 🤔

Reply:
It is moved to `init` because including auth header in requests should depend only on environment, which is set only once. You are right about not properly updated auth header value though. I mistakenly supposed that header will be updated along with this.token :man_facepalming: . I suggest to fix it by
```
this.httpClient.defaults.headers.common["Authentication"] = this.authHeader()
```
so that only value of the auth header will be updated, when token is changed, but the http client will not be reconfigured

Reply:
We could mention that tokens, as well as private key and identity token, should not be exposed, though I am not sure how it could help to secure them